pitop/common/configuration_file.py

```diff
@@ -0,0 +1,86 @@
+from pathlib import Path
+from re import compile
+
+
+def add_section(
+    filename: str, title: str, description: str, content: str, notes: str = ""
+) -> None:
+    """Add a section to a configuration file with the following format:
+
+    .. code-block:: python
+
+        # [<TITLE>] START
+        #
+        # <DESCRIPTION>
+        #
+        # Notes:
+        # <NOTES>
+        #
+        <CONTENT>
+        #
+        # [<APP_NAME>] END
+    """
+    text = f"""
+# [{title}] START
+#
+# {description}
+#
+# Notes:
+# {notes}
+#
+{content}
+#
+# [{title}] END"""
+    with open(filename, "a") as f:
+        f.write(text)
+
+
+def remove_section(filename: str, title: str) -> None:
+    """Removes a section of the provided title from the given filename. It
+    assumes that the format of the block to remove is the following:
+
+    .. code-block:: python
+
+        # [<TITLE>] START
+        #
+        # <DESCRIPTION>
+        #
+        # Notes:
+        # <NOTES>
+        #
+        <CONTENT>
+        #
+        # [<APP_NAME>] END
+    """
+    start_pattern = compile(rf"^(# \[{title}\] START)\n")
+    finish_pattern = compile(rf"^(# \[{title}\] END)\n")
+    delete_line = False
+
+    with open(filename, "r") as fr:
+        lines = fr.readlines()
+        with open(filename, "w") as fw:
```

Comment:
is it really necessary to have separate read & write handles? you can't use `with open(filename, "rw") as file:`?

Reply:
updated! I tried this before, but it just didn't work; adding `f.truncate()` after all the `write()` calls did it for me